connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```
@@ -1182,31 +1245,49 @@ public Void call() throws Exception {
-        if (started && initialState == TargetState.STARTED)
-            reconfigureConnectorTasksWithRetry(time.milliseconds(), connectorName);
+        final Callback<TargetState> onInitialStateChange = (error, newState) -> {
+            if (error != null) {
+                callback.onCompletion(new ConnectException("Failed to start connector: " + connectorName), null);
+                return;
+            }
 
-        return started;
+            // Use newState here in case the connector has been paused right after being created
+            if (newState == TargetState.STARTED) {
+                addRequest(
+                    new Callable<Void>() {
+                        @Override
+                        public Void call() {
+                            // Request configuration since this could be a brand new connector. However, also only update those
+                            // task configs if they are actually different from the existing ones to avoid unnecessary updates when this is
+                            // just restoring an existing connector.
+                            reconfigureConnectorTasksWithRetry(time.milliseconds(), connectorName);
+                            callback.onCompletion(null, null);
+                            return null;
+                        }
+                    },
+                    forwardErrorCallback(callback)
+                );
+            } else {
+                callback.onCompletion(null, null);
+            }
+        };
+        worker.startConnector(connectorName, configProps, ctx, this, initialState, onInitialStateChange);
     }
 
     private Callable<Void> getConnectorStartingCallable(final String connectorName) {
         return new Callable<Void>() {
             @Override
             public Void call() throws Exception {
                 try {
-                    startConnector(connectorName);
+                    startConnector(connectorName, (error, result) -> { });
```

This needs a log statement to avoid swallowing the exception silently.